repository/inventory_repo.go

```diff
@@ -152,11 +152,17 @@ func (i InventoryRepo) removeOldKeyData(ctx context.Context, key string) error {
 		var oldAssets map[string]string
 		err := i.cache.Get(ctx, k, &oldAssets)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to get stale assets for inventory key %q: %s", k, err)
```

Comment:
I think we'd expect them to exist if we have a key for them, but I can put a case in so that we don't error if the error isn't an `domain.ErrCacheNotFoundError` which would mean they'd been removed already. Also thinking about it, we probably shouldn't actually return here cause if there's more than one stale key and we error on the first one we won't attempt to cleanup the rest. So I'll change the returns to logs.